lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -259,6 +261,145 @@ class ExportCommand < HammerCLIKatello::SingleResourceCommand
+          if result['option_id'].nil? && result['option_name']
+            result['option_id'] = @command.resolver.content_view_id(result)
+          end
+          result
+        end
+      end
+
+      def option_sources
+        super << ContentViewIdParamSource.new(self)
+      end
+
+      def execute
+        cvv = content_view_version
+        repositories = cvv['repositories'].collect do |repo|
+          repository(repo['id'])
+        end
+
+        check_repo_download_policy(repositories)
+
+        repositories.each do |repo|
+          repo['packages'] = packages(repo['id'])
+          repo['errata'] = errata(repo['id'])
+        end
+
+        json = export_json(cvv, repositories)
+        create_tar(cvv, repositories, json)
+        return 0
+      end
+
+      def create_tar(cvv, repositories, json)
+        export_prefix = "export-#{cvv['id']}"
+        export_file = "#{export_prefix}.json"
+        export_repos_tar = "#{export_prefix}-repos.tar"
+        export_tar = "#{export_prefix}.tar"
+        published_repo_dir = "/var/lib/pulp/published/yum/https/repos/"
+
+        Dir.mkdir("#{options['option_export_dir']}/#{export_prefix}")
+
+        Dir.chdir(published_repo_dir) do
+          repositories.each do |repo|
+            repo_tar = "#{options['option_export_dir']}/#{export_prefix}/#{export_repos_tar}"
+            `tar cvfh #{repo_tar} #{repo['relative_path']}`
+          end
+        end
+
+        Dir.chdir("#{options['option_export_dir']}/#{export_prefix}") do
+          File.open(export_file, 'w') do |file|
+            file.write(JSON.pretty_generate(json))
+          end
+        end
+
+        Dir.chdir(options['option_export_dir']) do
+          `tar cf #{export_tar} #{export_prefix}`
+          FileUtils.rm_rf(export_prefix)
+        end
+      end
+
+      def content_view_version
+        resource_content_view_versions.call(:show, 'id' => options['option_id'])
+      end
+
+      def resource_content_view_versions
+        HammerCLIForeman.foreman_resource(:content_view_versions)
+      end
+
+      def repository(id)
+        HammerCLIForeman.foreman_resource(:repositories).call(:show, 'id' => id)
+      end
+
+      def packages(id)
+        HammerCLIForeman.foreman_resource(:packages).call(:index, 'repository_id' => id)['results']
+      end
+
+      def errata(id)
+        HammerCLIForeman.foreman_resource(:errata).call(:index, 'repository_id' => id)['results']
+      end
+
+      def check_repo_download_policy(repositories)
+        on_demand = repositories.select do |repo|
+          repo['download_policy'] == 'on_demand'
+        end
+        return true if on_demand.empty?
+
+        on_demand_names = repositories.collect { |repo| repo['name'] }
+        raise <<~MSG
+          All exported repositories must be set to an immediate download policy and re-synced.
+          The following repositories need action:
+            #{on_demand_names.join('\n')}
+        MSG
```

Comment:
@akofink @beav Here is the on_demand check. I used this style for the message to format the output.

Reply:
The way the code is written now, the additional step of re-publishing the content view is required. If we check the library repo's sync policy, that seems to be enough to have the packages found on the filesystem.

```rb
HammerCLIForeman.foreman_resource(:repositories).call(:show, 'id' => repo['library_instance_id'])['download_policy'] == 'on_demand'
```

You might think you'd need to use the library repo's `relative_path` to find the rpm on the filesystem, but it seems pulp only downloads an rpm once, so the symlink points to the same place as it's library repo's symlink.

Reply:
Good catch! And maybe this is cheating, but I see  other value as well -- https://github.com/Katello/katello/pull/7737

Reply:
In the end, it did not since we needed information from both.